pylearn2/datasets/dataset.py

```diff
@@ -18,7 +18,7 @@ def __iter__(self):
         return self.iterator()
 
```

Comment:
I hold myself to the same standard I am advocating here  e.g. I de-whitelisted utils/image.py after dwf raised concerns about PEP8 in one of my recent PRs ( https://github.com/lisa-lab/pylearn2/pull/1379 ) and the main point of that PR wasn't documentation.